framework/src/mesh/MooseMesh.C

```diff
@@ -1233,9 +1233,12 @@ MooseMesh::detectOrthogonalDimRanges(Real tol)
     }
   }
 
+  this->comm().max(max);
+  this->comm().min(min);
+
   _extreme_nodes.resize(8); // 2^LIBMESH_DIM
   // Now make sure that there are actual nodes at all of the extremes
-  unsigned int extreme_matches = 0;
+  std::vector<bool> extreme_matches(8, false);
```

Comment:
Hmm... actually it isn't, even though it should be: we don't currently have libMesh::Parallel::Communicator overloads for std::array.  I've finally got user unanimity on making libMesh C++11-only, but I haven't yet bit the bullet and broken C++98.